File: pset1/mario/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // GET HEIGHT
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while ((1 > height) || (8 < height));
    // SET PYRAMID SPEC INTEGERS
    int layer = 1;
    int hashes = 0;
    int spacesW = 0;
    int spacesH = 0;
    // BUILD PYRAMID
    do
    {
        //DECIDE HOW MANY SPACES NEEDED FOR LAYER
        spacesW = height - layer;
        // printf("height is %i; ",height);
        // printf("layer is %i; ",layer);
        // printf("will print %i",spacesW);
        // printf("will print %i",hashes);
        while (spacesH < spacesW)
        {
            //PRINT FIRST SPACES
            spacesH++;
            // printf("spacesH = %i",spacesH);
            printf(" ");
        }
        do
        {
            //PRINT FIRST HASHES
            printf("#");
            hashes++;
        }
        while (hashes < layer);
        hashes = 0;
        //PRINT 2 SPACES
        printf("  ");
        do
        {
            //PRINT HASHES
            printf("#");
            hashes++;
        }
        while (hashes < layer);
        //NEXT LINE

        printf("\n");

        //RESET INTEGERS FOR NEXT LAYER
        layer++;
        spacesW = 0;
        spacesH = 0;
        hashes = 0;
    }
    while (layer < (height + 1));

}

File: pset6/mario/more/mario.py
import cs50 as cs

height = 0

while height > 8 or height < 1:
    height = cs.get_int("Height: ")

completed_layers = 1

while completed_layers < height + 1:

    # initalize variables we will be using
    spacesH = 0
    hashes = 0
    spacesW = height - completed_layers
    
    # print first spaces
    while spacesH < spacesW:
        print(" ", end='')
        spacesH += 1
    # print first hashes
    while hashes != completed_layers:
        print("#", end='')
        hashes += 1
    # add two spaces
    print("  ", end='')
    # count down the right amount of hashes back to 0
    while hashes != 0:
        print("#", end='')
        hashes -= 1
    print("")

    completed_layers += 1

